internal/service/advancedclustertpf/resource_unit_test.go

```diff
@@ -0,0 +1,332 @@
+			no_table_scan                                                  = true
+			sample_refresh_interval_bi_connector                           = 310
+			sample_size_bi_connector                                       = 110
+			transaction_lifetime_limit_seconds                             = 300
+		}
+		`
+		fullUpdateResumed = strings.Replace(fullUpdate, "paused = true", "paused = false", 1)
+		vars              = map[string]string{
+			"groupId":     projectID,
+			"clusterName": clusterName,
+		}
+	)
+	shortenRetries()
+	testCase := resource.TestCase{
+		ProtoV6ProviderFactories: acc.TestAccProviderV6Factories,
+		Steps: []resource.TestStep{
+			{
+				Config: configBasic(projectID, clusterName, ""),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "state_name", "IDLE"),
+					resource.TestCheckResourceAttr(resourceName, "timeouts.create", "20s"),
+				),
+			},
+			{
+				Config: configBasic(projectID, clusterName, oneNewVariable),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "backup_enabled", "false"),
+					resource.TestCheckResourceAttr(resourceName, "state_name", "IDLE"),
+				),
+			},
+			{
+				Config: configBasic(projectID, clusterName, fullUpdate),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "state_name", "IDLE"),
+					resource.TestCheckResourceAttr(resourceName, "mongo_db_major_version", "8.0"),
+					resource.TestCheckResourceAttr(resourceName, "backup_enabled", "true"),
+					resource.TestCheckResourceAttr(resourceName, "paused", "true"),
+				),
+			},
+			{
+				Config: configBasic(projectID, clusterName, fullUpdateResumed),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "state_name", "IDLE"),
+					resource.TestCheckResourceAttr(resourceName, "backup_enabled", "true"),
+					resource.TestCheckResourceAttr(resourceName, "paused", "false"),
+				),
+			},
+			{
+				Config: configBasic(projectID, clusterName, fullUpdateResumed+advClusterConfig),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "mongo_db_major_version", "8.0"),
+					resource.TestCheckResourceAttr(resourceName, "advanced_configuration.change_stream_options_pre_and_post_images_expire_after_seconds", "100"),
+				),
+			},
+			{
+				ResourceName:                         resourceName,
+				ImportStateIdFunc:                    acc.ImportStateIDFuncProjectIDClusterName(resourceName, "project_id", "name"),
+				ImportState:                          true,
+				ImportStateVerify:                    true,
+				ImportStateVerifyIdentifierAttribute: "name",
+			},
+		},
+	}
+	unit.MockTestCaseAndRun(t, vars, &unit.MockHTTPDataConfig{AllowMissingRequests: true, AllowReReadGet: true}, &testCase)
+}
+
+func shortenRetries() {
+	advancedclustertpf.RetryMinTimeout = 1 * time.Second
```

Comment:
changing global vars 😱    i see how it helps here, but somebody could changed them from Prod code by mistake.  let's keep at the moment if there is not a safer way

Reply:
Would love to know if there is a better way.
Not worried about concurrent modifications as all of them want the same values so the end result is consistent.
Also, I don't believe we will run `Mock` and `Acc` tests together